studio.py

```diff
@@ -14,14 +22,211 @@
+                    app_service = APPS[self.app](
+                        proxy_service, self.certutil, self.app_path
+                    )
+                    app_service.start(recording["url"])
+
+                    if not recording.get("login", None):
+                        for i in range(0, RECORD_TIMEOUT):
+                            print(i)
+                            time.sleep(1)
+                    else:
+                        time.sleep(5)
+                        raw_input("Do user input and press <Return>")
+
+                    app_service.screen_shot(recording["screen_shot_path"])
+                    self.information["app_info"] = app_service.app_information()
+                    app_service.stop()
+
+            self.update_json_information(recording)
+            self.generate_zip_file(recording)
+            self.generate_manifest_file(recording)
+
+    def parse_sites_json(self):
+        print("Parsing sites json")
+        recordings = []
+        if self.sites is not None:
+            with open(self.sites, "r") as sites:
+                recordings_json = json.loads(sites.read())
+                self.information["app"] = self.app.lower()
+            if self.app in ["Firefox", "Chrome"]:
+                self.information["os_name"] = platform.system()
+                self.information["os_version"] = platform.release()
+                platform_name = platform.system().lower()
+
+            elif self.app in ["GeckoViewExample", "Fenix", "Fennec", "Refbrow"]:
+
+                device = ADBAndroid()
+                self.information["device_name"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["os_name"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["os_version"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+                if self.information["device_name"] == "Pixel 2":
+                    platform_name = "pixel2"
+                elif self.information["device_name"] == "Moto G (5)":
+                    platform_name = "motog5"
+
+            for recording in recordings_json:
+                label = recording.get("label", None)
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                        ]
+                    )
```

Comment:
Let's avoid this duplication by building the list and then appending the label if it's provided.

Reply:
This is much more concise as:

```python
name = [self.proxy, platform_name, self.app.lower(), site["domain"]]
if label:
    name.append(label)
name = "-".join(name)
```